framework/wazuh/core/rule.py

```diff
@@ -55,15 +56,17 @@ def check_status(status):
 
 def set_groups(groups, general_groups, rule):
     groups.extend(general_groups)
-    for g in groups:
+    for group in groups:
+        # These characters are the ones Core replaces when reading XML tags
+        group = re.sub('[\n\r ]+', '', group)
```

Comment:
This will replace ` ` anywhere it is found, converting for example `group name` into `groupname`. I did not find any existing group containing spaces, but after testing it myself, I see that this use case is accepted and could therefore be currently used by any user.

I modified rule 554 in `/var/ossec/ruleset/rules/0015-ossec_rules.xml` to add `  test syscheck ` group:
```XML
<rule id="554" level="5">
    <category>ossec</category>
    <decoded_as>syscheck_new_entry</decoded_as>
    <description>File added to the system.</description>
    <group>syscheck,  test syscheck ,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
```

Then I added a new rule in `local_rules.xml`:
```
 <rule id="100100" level="12">
    <if_group>test syscheck</if_group>
    <description>Test syscheck group</description>
</rule>
```

It is being triggered whenever I create a new file, so the space between `test` and `syscheck` is being kept. Indeed, if I remove it from the rule (`testsyscheck`) and restart, this error is shown:
```
2023/09/14 13:45:21 wazuh-analysisd: WARNING: (7610): Group 'testsyscheck' was not found. Invalid 'if_group'. Rule '100100' will be ignored.
```

**Conclusion:** Spaces should be removed before and after each group, but not inside it. Any number of spaces is accepted in the group (`test   syscheck` this is fine)